.gitignore

```diff
@@ -0,0 +1,88 @@
+# ------------------------------------------------------------------------------
+# List of file patterns we do not want in the repo.
+# ------------------------------------------------------------------------------
+
+# Object and library files
+*.[oa]
+*.so
+*.dylib
+
+# Backup files
+*~
+
+# Totalview breakpoint files
+*.TVD.*breakpoints
+
+# Mac related
+*.dSYM
+.DS_Store
+
+# Generated files in the root MFEM directory
+/deps.mk
+
+# Generated files in config/
+config/_config.hpp
+config/config.mk
+
+# Optional user configuration files in config/
+config/user.cmake
+config/user.mk
```

Comment:
I vote to exclude the `user*` files. They are optional, and it could be useful for `git status` to show them as a reminder that the repo was customized.